### DISCUSSION TITLE
Allow data-link to change local variables

### DISCUSSION THREAD

**Discussion by RenaudBompuis** [grafana/grafana#68429] [Dashboards, Panels & Visualizations]
I haven't found a way to make a data-link change a local variable.
Currently, a data-link will load a new page with variable parameters based on the item clicked.

Instead of loading a whole new page, I would propose a mechanism to define a data-link URL that updates a dashboard variable without requiring a whole reload, just like the user setting a variable in the dashboard header does.

This functionality would allow the use of dashboard variables in master-detail panel setups, where the user could click on a data-point or a table row defined with a data-link and simply see another panel update, just like selecting a new variable value works.

If this mechanism exists, I haven't been able to find how to define in from the documentation.

The data-link URL could make use of the `#` specifier in URLs, that allow reference to named sections within a document.

For instance, using a simple data-link defined as `#var-server=${server-name}` would see the local URL to be appended with `#var-server=colossus` for instance, without reloading the page.
There would need some mechanism to update the `server` variable, just as if the user has selected it from a dropdown in the UI.

I believe it would be a useful addition to the data-link paradigm, and make the user experience more dynamic.

Thank you.

Renaud

**Comment by RenaudBompuis** [comment]
Actually found that just using the standard data-link to rewrite the current URL seems to work.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82698. Feel free to continue the discussion around this there. Thank you!
